File: python_examples/exercises/randdna.pl
#!/usr/bin/perl
# Generate random DNA
#  using a random number generator to randomly select bases

use strict;
use warnings;

my $randdna=make_random_DNA(10000000);
#print $randdna."\n";
for (my $i=0; $i<length($randdna); $i+=10){
	$randdna =~ substr($randdna,$i,3,'XYZ'); 
}
#print $randdna."\n";
my $times = ($randdna =~ s/X/X/g);
print $times."\n";

sub make_random_DNA {

    # Collect arguments, declare variables
    my($length) = @_;

    my $dna;

    for (my $i=0 ; $i < $length ; ++$i) {

        $dna .= randomnucleotide(  );
    }

    return $dna;
}
sub randomnucleotide {

    my(@nucleotides) = ('A', 'C', 'G', 'T');

    # scalar returns the size of an array. 
    # The elements of the array are numbered 0 to size-1
    return randomelement(@nucleotides);
}

# randomelement
#
# randomly select an element from an array
#
# WARNING: make sure you call srand to seed the
#  random number generator before you call this function.

sub randomelement {

    my(@array) = @_;

    return $array[rand @array];
}

File: python_examples/exercises/randdna.py
import random
randdna=''
#for i in range(100000000):
#  randdna=randdna+'ACGT'[random.randint(0,3)]

randdna= ''.join(['ACGT'[random.randint(0,3)] for i in range(10000000)])
#print(randdna)
randdna=list(randdna)
for i in range(0,len(randdna),10):
  randdna[i:i+3]=list('XYZ')
randdna=''.join(randdna)
#print(randdna)
print(randdna.count('X'))
